casual/casual-caller/src/test/groovy/se/laz/casual/connection/caller/TransactionPoolMapperTest.groovy

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2022, The casual project. All rights reserved.
+ *
+ * This software is licensed under the MIT license, https://opensource.org/licenses/MIT
+ */
+
+package se.laz.casual.connection.caller
+
+
+import spock.lang.Shared
+import spock.lang.Specification
+
+import javax.transaction.Status
+import javax.transaction.Transaction
+
+class TransactionPoolMapperTest extends Specification
+{
+    @Shared
+    TransactionManagerImpl transactionManager
+
+    def setup()
+    {
+        TransactionPoolMapper.resetForTest()
+        TransactionPoolMapper.getInstance().setActiveForTest(true)
+        TransactionPoolMapper.getInstance().setTransactionManager(transactionManager = new TransactionManagerImpl())
+    }
+
+    def cleanupSpec()
+    {
+        TransactionPoolMapper.resetForTest()
+    }
+
+    def "Test default config state, should be disabled/inactive"()
+    {
+        setup:
+        TransactionPoolMapper.resetForTest()
+
+        expect:
+        !TransactionPoolMapper.getInstance().isPoolMappingActive()
+    }
+
+    def "Test with single transaction"()
+    {
+        setup:
+        Transaction transaction = new TransactionImpl(Status.STATUS_ACTIVE)
+        transactionManager.setCurrentTransaction(transaction)
+
+        String actualPoolName = "hello, world!"
+        TransactionPoolMapper.getInstance().setPoolNameForCurrentTransaction(actualPoolName)
+
+        expect:
+        TransactionPoolMapper.getInstance().getPoolNameForCurrentTransaction() == actualPoolName
+        TransactionPoolMapper.getInstance().getNumberOfTrackedTransactions() == 1
+    }
+
+    def "Test with many transactions"()
```

Comment:
Am I correct is saying that, this all relies on the Transaction implementations in wildfly and weblogic implementing equals and hashcode controls which actually utilise the transaction id somehow?
If so can we add tests which use those actual objects instead - include them as deps to testCompile for example, then we can actually test with those objects.

Reply:
Correct, the equality holds true for both and we need it to work like this. It should be possible to test with the actual Transaction-classes for WildFly at least where I should be able to pull in some Narayana dependency. Don't know about if I can get my hands on implementation classes  for WLS in some convenient manner, though. I'll have to look into it

Reply:
After looking at this it may be too much of a mess to build any useful tests on the WildFly specific classes. It quickly starts to devolve into Narayana implementation detail, different wrapper classes and so on. I'm not really sure these tests would be well invested time after gaining some more knowledge